clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -449,16 +465,24 @@ private static int parseAcks(String acksString) {
             if (callback != null)
                 callback.onCompletion(null, e);
             this.errors.record();
+            this.interceptors.onAcknowledgement(null, e);
             return new FutureFailure(e);
         } catch (InterruptedException e) {
             this.errors.record();
+            this.interceptors.onAcknowledgement(null, e);
             throw new InterruptException(e);
         } catch (BufferExhaustedException e) {
             this.errors.record();
             this.metrics.sensor("buffer-exhausted-records").record();
+            this.interceptors.onAcknowledgement(null, e);
             throw e;
         } catch (KafkaException e) {
             this.errors.record();
+            this.interceptors.onAcknowledgement(null, e);
+            throw e;
+        } catch (Exception e) {
```

Comment:
Does it make sense to call `errors.record()` here as well? If so, maybe we can remove the KafkaException case above since it would be doing the same thing.